apps/server/src/modules/h5p-editor/service/temporary-file-storage.service.ts

```diff
@@ -1,79 +1,110 @@
-import { ITemporaryFile, ITemporaryFileStorage, IUser } from '@lumieducation/h5p-server';
-import { Inject, Injectable, NotAcceptableException } from '@nestjs/common';
 import { S3ClientAdapter } from '@infra/s3-client';
+import { IFileStats, ITemporaryFile, ITemporaryFileStorage, IUser } from '@lumieducation/h5p-server';
+import {
+	HttpException,
+	Inject,
+	Injectable,
+	InternalServerErrorException,
+	NotAcceptableException,
+	NotFoundException,
+} from '@nestjs/common';
+import { ErrorUtils } from '@src/core/error/utils';
 import { ReadStream } from 'fs';
 import { Readable } from 'stream';
 import { H5pFileDto } from '../controller/dto/h5p-file.dto';
-import { H5pEditorTempFile } from '../entity/h5p-editor-tempfile.entity';
 import { H5P_CONTENT_S3_CONNECTION } from '../h5p-editor.config';
-import { TemporaryFileRepo } from '../repo/temporary-file.repo';
 
 @Injectable()
 export class TemporaryFileStorage implements ITemporaryFileStorage {
-	constructor(
-		private readonly repo: TemporaryFileRepo,
-		@Inject(H5P_CONTENT_S3_CONNECTION) private readonly s3Client: S3ClientAdapter
-	) {}
+	constructor(@Inject(H5P_CONTENT_S3_CONNECTION) private readonly s3Client: S3ClientAdapter) {}
 
 	private checkFilename(filename: string): void {
-		if (!/^[a-zA-Z0-9/._-]+$/g.test(filename) && filename.includes('..') && filename.startsWith('/')) {
-			throw new NotAcceptableException(`Filename contains forbidden characters or is empty: '${filename}'`);
+		filename = filename.split('.').slice(0, -1).join('.');
```

Comment:
are you sure about that?

Reply:
Check has been taken directly from the ContentStorage. Lumi modifies the file-extension so we should ignore it.